Tasks/AzureRmWebAppDeployment/azurermwebappcontainerdeployment.ts

```diff
@@ -0,0 +1,65 @@
+import tl = require('vsts-task-lib/task');
+import util = require('util');
+
+var azureRESTUtility = require ('azurerest-common/azurerestutility.js');
+var parameterParser = require("./parameterparser.js").parse;
+
+export async function deployWebAppImage(endPoint, resourceGroupName, webAppName) {
+    var startupCommand = tl.getInput('StartupCommand', false);
+    var appSettings = tl.getInput('AppSettings', false);
+    var imageSourceAndTag;
+
+    // Construct the image
+    var dockerNamespace = tl.getInput('DockerNamespace', true);
+    var dockerRepository = tl.getInput('DockerRepository', true);
+    var dockerImageTag = tl.getInput('DockerImageTag', false);
+
+    if(dockerImageTag) {
+        imageSourceAndTag = dockerNamespace + "/" + dockerRepository + ":" + dockerImageTag;          
+    } else {
+        imageSourceAndTag = dockerNamespace + "/" + dockerRepository;
+    }
+
+    if(imageSourceAndTag)
+    {
+        tl.debug("Deploying the image " + imageSourceAndTag + " to the webapp " + webAppName);
+        
+        appSettings = appSettings ? appSettings.trim() : "";        
+        appSettings = "-DOCKER_CUSTOM_IMAGE_NAME " + imageSourceAndTag + " " + appSettings;
+    
+        // Update webapp application setting
+        var webAppSettings = await azureRESTUtility.getWebAppAppSettings(endPoint, webAppName, resourceGroupName, false, null);
```

Comment:
The value of slot will be persisted  if user selects a usual webapp before selecting the linux one. So, to handle such situations, I am sending false. Will update it with slotId, when we support slots.